l10n_ec_withhold/tests/test_l10n_ec_sale_withhold.py

```diff
@@ -0,0 +1,362 @@
+            line.tax_group_withhold_id = tax_group
+            line.tax_withhold_id = taxes
+
+        with self.assertRaises(UserError):
+            wizard2.save().button_validate()
+
+    def test_l10n_ec_invoice_exist_in_withhold(self):
+        """
+        Fail when other invoice exist in different withhold
+        """
+        partner = self.partner_with_email
+        invoice = self.get_invoice(partner)
+
+        wizard1 = Form(
+            self.env["l10n_ec.wizard.create.sale.withhold"].with_context(
+                **{"active_ids": invoice.ids}
+            )
+        )
+
+        wizard1.issue_date = invoice.invoice_date
+        wizard1.journal_id = invoice.journal_id
+        wizard1.electronic_authorization = "1111111111"
+        wizard1.document_number = "1-1-1"
+        tax_group, taxes = self.get_tax_group_and_taxes()
+
+        with wizard1.withhold_line_ids.new() as line:
+            line.invoice_id = invoice
+            line.tax_group_withhold_id = tax_group
+            line.tax_withhold_id = taxes
+
+        # Create first withhold
+        wizard1.save().button_validate()
+
+        # Create other withhold with the same invoice
+        wizard2 = Form(
+            self.env["l10n_ec.wizard.create.sale.withhold"].with_context(
+                **{"active_ids": invoice.ids}
+            )
+        )
+
+        wizard2.issue_date = invoice.invoice_date
+        wizard2.journal_id = invoice.journal_id
+        wizard2.electronic_authorization = "1111111111"
+        wizard2.document_number = "1-1-2"
+        tax_group, taxes = self.get_tax_group_and_taxes()
+
+        with wizard2.withhold_line_ids.new() as line:
+            line.invoice_id = invoice
+            line.tax_group_withhold_id = tax_group
+            line.tax_withhold_id = taxes
+
+        with self.assertRaises(UserError):
+            wizard2.save().button_validate()
+
+    def test_l10n_ec_fail_when_invoice_is_paid(self):
+        """
+        Fail when invoice is paid
+        """
+        partner = self.partner_with_email
+        invoice = self.get_invoice(partner)
+
+        invoice.write({"payment_state": "paid"})
+
+        with self.assertRaises(UserError):
+            Form(
+                self.env["l10n_ec.wizard.create.sale.withhold"].with_context(
+                    **{"active_ids": invoice.ids}
+                )
+            )
+
+    def get_tax_group_and_taxes(self):
+        """
+        Create line for withhold
+        """
+        tax_group = self.env["account.tax.group"].search(
+            [("l10n_ec_type", "=", "withhold_vat")]
+        )
+
+        taxes = self.env["account.tax"].search(
+            [
+                ("company_id", "=", self.company.id),
+                ("tax_group_id", "=", tax_group.id),
+                ("type_tax_use", "=", "sale"),
+                ("amount", "=", -100),
+            ]
+        )
+
+        return tax_group, taxes
+
+    def get_invoice(self, partner):
```

Comment:
Hice un cambio para desactivar toda conexion con el SRI agregando un decorador en la funcion, asi no se necesitaria hacer un patch y queda un poco mas limpio el codigo
Mira el commit https://github.com/Odoo-EC/l10n-ecuador/commit/e62bd1e47e947070acd099bd98a852894971987a

```suggestion
    @patch_service_sri
    def get_invoice(self, partner):
```